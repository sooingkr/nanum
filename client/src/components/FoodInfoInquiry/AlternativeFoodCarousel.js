import React from 'react';
import { PropTypes } from 'prop-types';

import { isArray } from 'lodash';

import Carousel from '../Common/Carousel';
import AlternativeFoodItem from './AlternativeFoodItem';

const AlternativeFoodCarousel = ({foods}) => {

    const foodCarouselSettings = {
      arrows: true,
      dots: false,
      infinite: true,
      speed: 500,
      slidesToShow: 2,
      slidesToScroll: 2,
      responsive: [
        { breakpoint: 1025, settings: { slidesToShow: 1, slidesToScroll: 1 }},
        { breakpoint: 769, settings: { slidesToShow: 2, slidesToScroll: 2 }},
        { breakpoint: 480, settings: { slidesToShow: 2, slidesToScroll: 2, vertical: true}}]
    };

  return (
      <Carousel className='food-carousel__sliders' settings={foodCarouselSettings}>
        {
          isArray(foods) && foods.map((food, index) =>
            <div className="food-carousel__item" key={ index }>
              <AlternativeFoodItem food={food}/>
            </div>
          )
        }
      </Carousel>
    );
}

AlternativeFoodCarousel.propTypes = {
  foods: PropTypes.array
}

export default AlternativeFoodCarousel;