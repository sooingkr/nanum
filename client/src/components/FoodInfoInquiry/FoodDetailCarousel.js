import React from 'react';
import { PropTypes } from 'prop-types';

import AlternativeFoodCarousel from "../../components/FoodInfoInquiry/AlternativeFoodCarousel";

const FoodDetailCarousel = ({foods}) => {

  return (
    <div className='alternative-food'>
      <AlternativeFoodCarousel foods={foods}/>
    </div>
  );
}

FoodDetailCarousel.propTypes = {
  foods: PropTypes.array.isRequired
}

export default FoodDetailCarousel;