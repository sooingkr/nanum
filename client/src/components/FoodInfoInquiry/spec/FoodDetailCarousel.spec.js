import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FoodDetailCarousel from '../../../components/FoodInfoInquiry/FoodDetailCarousel';
import AlternativeFoodCarousel from "../AlternativeFoodCarousel";

describe('FoodDetailCarousel component', () => {
  const wrapper = shallow(<FoodDetailCarousel/>);

  it('should render <FoodDetailCarousel /> components', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render div className="alternative-food"', () => {
    expect(wrapper.find(".alternative-food").length).toBe(1);
  });

  describe('AlternativeFoodCarousel component with props', () => {
    it('should render <AlternativeFoodCarousel props/>', () => {
      expect(wrapper.find(AlternativeFoodCarousel).prop('foods').length).toBeGreaterThanOrEqual(1);
    });
  });

});