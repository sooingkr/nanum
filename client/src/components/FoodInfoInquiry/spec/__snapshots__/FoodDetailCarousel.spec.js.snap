// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodDetailCarousel component should render <FoodDetailCarousel /> components 1`] = `
<div
  className="alternative-food"
>
  <AlternativeFoodCarousel />
</div>
`;
